### ISSUE TITLE
Inline style and CSP policy

### ISSUE THREAD

**Issue by sasha-3ap** [ricardo-ch/react-easy-crop#139]
When CSP security policy with directive style-src is set, unless 'unsafe-inline' value is used, react-easy-crop inline style are ignored which results in incorrect react-easy-crop render. As setting 'unsafe-inline' defeats purpose of CSP, react-easy-crop should provide solution without inline styles. Please check '[Disable inline style](https://dontkry.com/posts/code/disable-inline-styles.html)' from more information and possible solution.

If I solve this I will create PR.

**Comment by ValentinH** [comment]
You are already able to request the CSS explicitly: see https://github.com/ricardo-ch/react-easy-crop/blob/master/README.md#styles.

Does it cover your use case?

**Comment by sasha-3ap** [comment]
Seems it does not. Even if we use disableAutomaticStylesInjection and css it is still [added as inline style](https://github.com/ricardo-ch/react-easy-crop/blob/master/src/index.tsx#L106). In render we have inline style [here](https://github.com/ricardo-ch/react-easy-crop/blob/master/src/index.tsx#L467), [here](https://github.com/ricardo-ch/react-easy-crop/blob/master/src/index.tsx#L484) and  [here](https://github.com/ricardo-ch/react-easy-crop/blob/master/src/index.tsx#L494). 

In Material UI CSP with nonce is [supported](https://material-ui.com/styles/advanced/#how-does-one-implement-csp) for inline styles by reading nonce from meta tag. Similar solution could be used in react-easy-crop, add nonce from meta tag. Or add style through javascript. Or ...

**Comment by ValentinH** [comment]
I understand for the inline styles in render. However, the styles injected in the `head` are not if you set `disableAutomaticStylesInjection` to true.

Just to understand, is this issue a regression of version 3.0?

**Comment by sasha-3ap** [comment]
I tried adding disableAutomaticStylesInjection to Cropper and import css in component that used it. Is this a proper way? I would still error:
![image](https://user-images.githubusercontent.com/54268606/82900368-b58dd400-9f5c-11ea-86ba-20e2ac325599.png)


Version is latest. Not sure if this can be considered as regression.  

We generate nonce on every new application request but we also insert nonce in meta tag (as required by MUI) so it can be used in inline styles.

**Comment by ValentinH** [comment]
What's the `injectStylesIntoStyleTag` function?

**Comment by sasha-3ap** [comment]
It is from style loader used by webpack, [injectStylesIntoStyleTag](https://github.com/webpack-contrib/style-loader/blob/master/src/runtime/injectStylesIntoStyleTag.js). It goes over styles and if configured adds nonce as well.

**Comment by ValentinH** [comment]
So isn't the error due to a misconfiguration in webpack? Could you please try version 2.X?

**Comment by sasha-3ap** [comment]
There is no misconfiguration in webpack, however I found that webpack did not use __webpack_nonce__ to set nonce for inline styles. I opened bug for it but due the difficult setup to show the bug I am not sure it will get proper attention. In the meantime, since Material UI is already used I explored how it uses nonce from meta tag and applied similar principal to this case. 

Since MUI used JSS solution was to introduces css required by react-easy-crop using @global keyword.
